agents-api/agents_api/queries/agents/create_agent.py

```diff
@@ -25,51 +25,78 @@
 ModelT = TypeVar("ModelT", bound=Any)
 T = TypeVar("T")
 
-
-@rewrap_exceptions(
-    {
-        psycopg_errors.ForeignKeyViolation: partialclass(
-            HTTPException,
-            status_code=404,
-            detail="The specified developer does not exist.",
-        ),
-        psycopg_errors.UniqueViolation: partialclass(
-            HTTPException,
-            status_code=409,
-            detail="An agent with this canonical name already exists for this developer.",
-        ),
-        psycopg_errors.CheckViolation: partialclass(
-            HTTPException,
-            status_code=400,
-            detail="The provided data violates one or more constraints. Please check the input values.",
-        ),
-        ValidationError: partialclass(
-            HTTPException,
-            status_code=400,
-            detail="Input validation failed. Please check the provided data.",
-        ),
-        TypeError: partialclass(
-            HTTPException,
-            status_code=400,
-            detail="A type mismatch occurred. Please review the input.",
-        ),
-    }
+raw_query = """
```

Comment:
I dont think this is completed yet, because it doesn't create `agent_default_settings`. Maybe we can merge this PR and pick that up after